tests/unit/lms/services/d2l_api/client_test.py

```diff
@@ -192,6 +194,26 @@ def test_public_url(self, svc, basic_client):
 
         assert public_url == basic_client.api_url.return_value
 
+    @pytest.mark.parametrize(
+        "status_code,exception,error_code",
+        [
+            (404, FileNotFoundInCourse, "d2l_file_not_found_in_course"),
+            (400, ExternalRequestError, None),
+        ],
+    )
+    def test_public_url_raises(
+        self, svc, basic_client, status_code, exception, error_code
+    ):
+        basic_client.request.side_effect = ExternalRequestError(
+            response=factories.requests.Response(status_code=status_code)
+        )
+
+        with pytest.raises(exception) as excinfo:
+            svc.public_url("COURSE_ID", "FILE_ID")
+
+        if error_code:
```

Comment:
Parametrized this, it does have this conditional wrinkle in it.

Reply:
Ack! Can't win 'em all.